test/handler.test.js

```diff
@@ -194,14 +194,38 @@ t.test('handle', async t => {
 
     await handle({ context: contextSpy, logger: loggerSpy, batchSize: 1 })
 
-    t.equal(connectionSpy.send.callCount, 2) // Only two sends in the 4 block messages
+    t.equal(connectionSpy.send.callCount, 0) // Blocks requested and cancelled in same message
```

Comment:
The mock actually intercepts the HTTP request from client, and given async operation means now we actually cancel everything 🎉